app/views/editor/rich.html.erb

```diff
@@ -483,6 +493,54 @@
     $("#obscureLocation").prop('checked', <%= @map_blurred %>);
     editor.mapModule.blurredLocation.setBlurred(<%= @map_blurred %>);
 
+
+
+  setTimeout(function(){
+
+    // settings at https://github.com/ichord/At.js/wiki/Base-Document#settings
+    var at_config = {
+        at: "@",
+        delay: 400,
+        callbacks: {
+        remoteFilter: debounce(function(query, callback) {
+            $.getJSON("/api/srch/profiles?query=" + query + "&sort_by=recent&field=username", {}, function(data) {
+            if (data.hasOwnProperty('items') && data.items.length > 0) {
+                callback(data.items.map(function(i) { return i.doc_title }));
+            }
+            });
+            }, 200)
+        },
+        limit: 20
+        },
+        hashtags_config = {
+        at: "#",
+        delay: 400,
+        callbacks: {
+            remoteFilter: debounce(function(query, callback) {
+            if (query != ''){
+                $.post('/tag/suggested/' + query, {}, function(response) {
+                callback(response.map(function(tagnames){ return tagnames }));
+                });
+                }
+            }, 200)
+            },
+        limit: 20
+        },
+        emojis_config = {
+        at: ':',
+        delay: 400,
+        data: Object.keys(emoji).map(function(name){ return {'name': name, 'value': emoji[name]}}),
+        displayTpl: "<li>${value} ${name}</li>",
+        insertTpl: ":${name}:",
+        limit: 100
+    }
+
+    $('.wk-wysiwyg')
+        .atwho(hashtags_config)
+        .atwho(emojis_config)
+        //.atwho(at_config) // off for performance reasons for now
```

Comment:
@TildaDares here it is! This is where I turned off the username autocompletion. It uses the same API backend as your code. But the front-end implementation may be a little different as it's within a rich text editor!